pkg/kubeinteraction/secrets.go

```diff
@@ -65,5 +66,14 @@ func (k Interaction) UpdateSecretWithOwnerRef(ctx context.Context, logger *zap.S
 
 func (k Interaction) CreateSecret(ctx context.Context, ns string, secret *corev1.Secret) error {
 	_, err := k.Run.Clients.Kube.CoreV1().Secrets(ns).Create(ctx, secret, metav1.CreateOptions{})
-	return err
+	if err != nil {
```

Comment:
I don't think this is the correct abstraction layer to swallow the AlreadyExists error. Functionally this method is only used when we create a basicAuth secret for a pipelinerun, but this is a generic package. Do you mind moving this check to `PacRun.startPR`? The other benefit of having this check at that level is we can emit the warning as an event on the repository CR which is more useful than just logging it